ITEventsTests/SearchParametersPresenterTests/SearchParametersPresenterTests.swift

```diff
@@ -0,0 +1,121 @@
+        searchParametersService.otherTags = [Tag(id: 3, name: "Android"),
+                                             Tag(id: 4, name: "Python")]
+        //When
+        searchParametersPresenter.setup()
+        
+        //Then
+        XCTAssertEqual(view.selectedTags, searchParametersService.selectedTags)
+        XCTAssertEqual(view.otherTags, searchParametersService.otherTags)
+        XCTAssertEqual(view.fillCallsCount, 1)
+    }
+    
+    func testPresenterSelectSearchTagAndSave() {
+        //Given
+        searchParametersService.selectedTags = [Tag(id: 1, name: "JavaScript"),
+                                                Tag(id: 2, name: "iOS")]
+        searchParametersService.otherTags = [Tag(id: 3, name: "Android"),
+                                             Tag(id: 4, name: "Python")]
+        
+        //When
+        searchParametersPresenter.setup()
+        searchParametersPresenter.selectTag(Tag(id: 3, name: "Android"))
+        searchParametersPresenter.saveSettings()
+        
+        //Then
+        let expectedSelectedTags = [Tag(id: 3, name: "Android"),
+                                 Tag(id: 1, name: "JavaScript"),
+                                 Tag(id: 2, name: "iOS")]
+        let expectedOtherTags = [Tag(id: 4, name: "Python")]
+        XCTAssertEqual(searchParametersService.selectedTags, expectedSelectedTags)
+        XCTAssertEqual(searchParametersService.otherTags, expectedOtherTags)
+        XCTAssertEqual(view.fillCallsCount, 1)
+    }
+    
+    func testPresenterDeselectSearchTagAndSave() {
+        //Given
+        searchParametersService.selectedTags = [Tag(id: 1, name: "JavaScript"),
+                            Tag(id: 2, name: "iOS")]
+        searchParametersService.otherTags = [Tag(id: 3, name: "Android"),
+                         Tag(id: 4, name: "Python")]
+        
+        //When
+        searchParametersPresenter.setup()
+        searchParametersPresenter.deselectTag(Tag(id: 1, name: "JavaScript"))
+        searchParametersPresenter.saveSettings()
+        
+        //Then
+        let expectedSelectedTags = [Tag(id: 2, name: "iOS")]
+        let expectedOtherTags = [Tag(id: 3, name: "Android"),
+                              Tag(id: 1, name: "JavaScript"),
+                              Tag(id: 4, name: "Python")]
+        XCTAssertEqual(searchParametersService.selectedTags, expectedSelectedTags)
+        XCTAssertEqual(searchParametersService.otherTags, expectedOtherTags)
+        XCTAssertEqual(view.fillCallsCount, 1)
+    }
+    
+    func testPresenterSelectNonExistingSearchTagAndSave() {
+        //Given
+        let selectedTags = [Tag(id: 1, name: "JavaScript"),
+                            Tag(id: 2, name: "iOS")]
+        let otherTags = [Tag(id: 3, name: "Android"),
+                         Tag(id: 4, name: "Python")]
+        searchParametersService.updateTags(selectedTags: selectedTags, otherTags: otherTags)
```

Comment:
1. Since ```searchParametersService``` is mock in this file (we are testing the presenter) we should not use ```updateTags``` as we simply can use the properties:
```swfit
searchParametersService.selectedTags = [Tag(id: 1, name: "JavaScript"), ...
searchParametersService.otherTags = [Tag(id: 3, name: "Android"), ...
```

2. It'd better to postfix all mock variables/properties with "mock" word.

So for example instead of 
```swift
var searchParametersService: SearchParametersServiceMock!
```
should be 
```swift
var searchParametersServiceMock: SearchParametersServiceMock!
```
3. After we postfix all the mock variables/properties make sure arrays from 1. (selectedTags and otherTags) are still aligned.

Reply:
I need ```selectedTags``` and ```otherTags``` as variables in this test because I compare with them in ```XCTAssertEqual```